File: Matrix Audio/Matrix.Audio.Models/Promo.cs
namespace Matrix.Audio.Models;
public class Promo : Entity
{
    public required string Code { get; set; }
    public required decimal Min { get; set; }
    public required decimal Discount { get; set; }
    public required string DiscountType { get; set; } // e.g., "percentage" or "fixed"
    public required DateTime ValidFrom { get; set; }
    public required DateTime ValidTo { get; set; }
    public required DateTime DateCreated { get; set; } = DateTime.Now;
    public bool IsActive { get; set; }
}

File: Matrix Audio/matrix.audio.client/src/app/models/promo.ts
import { Entity } from "./entity";
import { DiscountType } from "./subscription-plan";

export interface Promo extends Entity {
    code: string;
    min: number;
    discount: number;
    discountType: DiscountType; // percentage or fixed
    validFrom: string;
    validTo: string;
    dateCreated: Date;
    isActive: boolean;
}